storage/remote/otlptranslator/prometheusremotewrite/helper_test.go

```diff
@@ -52,13 +52,16 @@ func TestCreateAttributes(t *testing.T) {
 
 	testCases := []struct {
 		name                      string
-		promoteResourceAttributes []string
+		promoteResourceAttributes *PromoteResourceAttributes
```

Comment:
You are changing the nature of the existing test cases, that's not a goal of the PR. Instead, add test case fields for the new functionality:
```suggestion
		promoteAllResourceAttributes bool
		promoteResourceAttributes []string
		ignoreResourceAttributes     []string
```